scripts/ssl_tts/make_supdata.py

```diff
@@ -0,0 +1,434 @@
+                content_embedding = content_embedding.t()
+                content_log_probs = content_log_probs[: encoded_len.item()]
+                content_log_probs = content_log_probs.t()
+                content_probs = torch.exp(content_log_probs)
+
+                duration = torch.ones(content_embedding.shape[1]) * 4.0
+
+                bsi_start = segment_indices[idx][0]
+                bsi_end = segment_indices[idx][1]
+                speaker_embedding = torch.mean(batch_speaker_embeddings[bsi_start : bsi_end + 1], dim=0)
+
+                l2_norm = torch.norm(speaker_embedding, p=2)
+                speaker_embedding = speaker_embedding / l2_norm
+
+                if args.ssl_content_emb_type == "probs":
+                    final_content_embedding = content_probs
+                elif args.ssl_content_emb_type == "embedding":
+                    final_content_embedding = content_embedding
+                elif args.ssl_content_emb_type == "log_probs":
+                    final_content_embedding = content_log_probs
+                elif args.ssl_content_emb_type == "embedding_and_probs":
+                    final_content_embedding = torch.cat([content_embedding, content_probs], dim=0)
+
+                if args.use_unique_tokens == 1:
+                    token_predictions = torch.argmax(content_probs, dim=0)
+                    # print("token predictions:", token_predictions)
+                    content_buffer = [final_content_embedding[:, 0]]
+                    unique_content_embeddings = []
+                    unique_tokens = []
+                    durations = []
+                    for _t in range(1, final_content_embedding.shape[1]):
+                        if token_predictions[_t] == token_predictions[_t - 1]:
+                            content_buffer.append(final_content_embedding[:, _t])
+                        else:
+                            durations.append(len(content_buffer) * 4)
+                            unique_content_embeddings.append(torch.mean(torch.stack(content_buffer), dim=0))
+                            content_buffer = [final_content_embedding[:, _t]]
+                            unique_tokens.append(token_predictions[_t].item())
+
+                    if len(content_buffer) > 0:
+                        durations.append(len(content_buffer) * 4)
+                        unique_content_embeddings.append(torch.mean(torch.stack(content_buffer), dim=0))
+                        unique_tokens.append(token_predictions[_t].item())
+
+                    unique_content_embedding = torch.stack(unique_content_embeddings)
+                    final_content_embedding = unique_content_embedding.t()
+                    duration = torch.tensor(durations).float()
+                    # print("duration ds", duration)
+                    encoded_len = torch.tensor(final_content_embedding.shape[1]).long()
+
+                mel_len = int(batch['audio_len'][idx].item() / 256.0)
```

Comment:
Magic number here--what's 256? Should this be `hop_length`?

Reply:
Yes, made it configurable.